modules/docs/markdown/03-protocols/02-definition.md

```diff
@@ -0,0 +1,106 @@
+
+#### Summary
+
+In summary, TCP/IP is a collection of protocols, grouped into layers, that make up the basis for interactions between computers.
+
+## HTTP
+
+Before we move on, let's make sure we have a common understanding of what HTTP actually is. HTTP stands for HyperText Transfer Protocol. As mentioned above, it builds upon TCP to send messages between two hosts. The most important thing for us to understand here is what HTTP defines and what is does not define.
+
+HTTP defines semantics for communicating between host applications in a request-response model. We are all likely familiar with making an HTTP request and getting back a response. The most basic example of this is loading a web page on the internet.
+
+Below is a collection of things that HTTP **DOES** define. Note that we are not going to be exhaustive in our definition of HTTP. The below is just to give an overview.
+
+#### Request and Response Headers
+
+HTTP defines that each request and response may have headers which contain additional information related to the request or response. These are key-value pairs where the key is case-insensitive.
+
+#### Request Methods
+
+Requests have methods such as `GET`, `PUT`, `POST`, and `DELETE` (there are currently 9 total). These methods are intended to inform the destination application what the requester wishes to do with a given resource.
+
+#### Response Status Codes
+
+Responses in HTTP come with an integer status code that represents the category of the result of the request. For example a status code between 200 and 299 is used to indicate that a request was successful.
+
+#### Message Body
+
+HTTP defines that requests and responses each have an optional message body. This is usually where the majority of information is contained.
+
+#### Things NOT Defined in HTTP
+
+Now that we have covered the basis of what HTTP defines, it is important to call out a few things that it does not define.
+
+The most major thing left out of the HTTP specification is the format for data in the body of requests and responses. As far as HTTP is concerned, these bodies are just Bytes. Defining what these bodies look like is left to higher-level protocols (as we will discuss below).
+
+HTTP also does not define what types of resources requests should be able to work with. This means requests can be used to get or modify files stored in a system, in memory state, a third party API, or anything else one could imagine.
+
+Additionally, although HTTP provides some mechanisms for basic authentication, it does not define anything around authorization. Other protocols, such as OAuth 2.0 are often used with HTTP applications.
+
+#### REST vs HTTP
```

Comment:
I'm putting this comment here in a semi-arbitrary fashion, but I think we're missing a transition section after `things not defined in http` that we'd call `protocols in the level of smithy`, where we'd exemplify it concretely : 

"a protocol is a set of concrete rules that define how data modelled in smithy get transcribed into lower-level semantics. It can be accompanied by a set of annotations that allow to capture the details of that transcription that cannot be inferred from the shape themselves, or to customise that transcription" 

then give a concrete example of the @http annotation for instance.

Reply:
Addressed in https://github.com/disneystreaming/smithy4s/pull/719/commits/7116e1fa736765210ed008ec027a9109ed6d61a7